contracts/governance/Governor.sol

```diff
@@ -4,208 +4,224 @@ pragma solidity ^0.8.0;
 
 import "../utils/cryptography/ECDSA.sol";
 import "../utils/cryptography/draft-EIP712.sol";
+import "../utils/introspection/ERC165.sol";
 import "../utils/Address.sol";
 import "../utils/Context.sol";
-import "../utils/Time.sol";
+import "../utils/Timers.sol";
 import "./IGovernor.sol";
 
 /**
  * @dev Core of the governance system, designed to be extended though various modules.
  *
- * _Available since v4.2._
+ * _Available since v4.3._
  */
-abstract contract Governor is IGovernor, EIP712, Context {
-    using Time for Time.Timer;
+abstract contract Governor is Context, ERC165, EIP712, IGovernor {
+    using Timers for Timers.BlockNumber;
 
-    bytes32 private constant _BALLOT_TYPEHASH = keccak256("Ballot(uint256 proposalId,uint8 support)");
+    bytes32 public constant BALLOT_TYPEHASH = keccak256("Ballot(uint256 proposalId,uint8 support)");
 
-    struct Proposal {
-        Time.Timer timer;
-        uint256 snapshot;
+    struct ProposalCore {
+        Timers.BlockNumber voteStart;
+        Timers.BlockNumber voteEnd;
+        bool executed;
         bool canceled;
     }
 
     string private _name;
-    string private _version;
 
-    mapping (uint256 => Proposal) private _proposals;
-    mapping (uint256 => mapping (address => bool)) private _votes;
+    mapping(uint256 => ProposalCore) private _proposals;
 
     /**
      * @dev Sets the value for {name} and {version}
      */
-    constructor(string memory name_, string memory version_)
-    EIP712(name_, version_)
-    {
+    constructor(string memory name_) EIP712(name_, version()) {
         _name = name_;
-        _version = version_;
+    }
+
+    /**
+     * @dev See {IERC165-supportsInterface}.
+     */
+    function supportsInterface(bytes4 interfaceId) public view virtual override(IERC165, ERC165) returns (bool) {
+        return interfaceId == type(IGovernor).interfaceId || super.supportsInterface(interfaceId);
     }
 
     /**
      * @dev See {IGovernor-name}.
      */
-    function name() external view virtual override returns (string memory) {
+    function name() public view virtual override returns (string memory) {
         return _name;
     }
 
     /**
-     * @dev See {IGovernor-version}.
+     * @dev See {IGovernor-vestion}.
+     */
+    function version() public view virtual override returns (string memory) {
+        return "1";
+    }
+
+    /**
+     * @dev See {IGovernor-hashProposal}.
```

Comment:
Should we also document the default hash function in case people want to recreate it outside the contract?